File: src/main/resources/static/js/room/Rooms.js
import {Room} from "./Room.js";

export class Rooms {
  #rooms = []

  constructor(roomDtos) {
    for (let i = 0; i < roomDtos.length; i++) {
      this.#rooms.push(new Room(roomDtos[i]));
    }
  }

  get rooms() {
    return this.#rooms;
  }
}

File: src/main/java/chess/domain/game/room/Rooms.java
package chess.domain.game.room;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Rooms {

    private final List<Room> rooms;

    public Rooms(final List<Room> rooms) {
        this.rooms = new ArrayList<>(rooms);
    }

    public List<Room> toList() {
        return Collections.unmodifiableList(rooms);
    }

}
